# HG changeset patch
# User Richard Russon <rich@flatcap.org>
# Date 1461202338 -3600
#      Thu Apr 21 02:32:18 2016 +0100
# Branch gentoo-1.7
# Node ID 8867916fa2e2e36d8653b7c1ce53af635aa2bc71
# Parent  885f1d7126a4ffa099f9c0d8a311c1c7e9f2f74a
[PATCH] sidebar_on_right
From 65df1c8cb5d34ccf8942b05e869b3cf5cdf73f0e Mon Sep 17 00:00:00 2001
---
 curs_lib.c |  8 ++++++--
 init.h     |  5 +++++
 mutt.h     |  1 +
 sidebar.c  | 33 +++++++++++++++++++++++++--------
 4 files changed, 37 insertions(+), 10 deletions(-)

diff -r 885f1d7126a4 -r 8867916fa2e2 PATCHES
--- a/PATCHES	Tue Jan 26 18:31:05 2016 +0000
+++ b/PATCHES	Thu Apr 21 02:32:18 2016 +0100
@@ -1,3 +1,4 @@
+patch-sidebar-on-right-neomutt
 patch-compress-neomutt
 patch-cond-date-neomutt
 patch-nested-if-neomutt
diff -r 885f1d7126a4 -r 8867916fa2e2 curs_lib.c
--- a/curs_lib.c	Tue Jan 26 18:31:05 2016 +0000
+++ b/curs_lib.c	Thu Apr 21 02:32:18 2016 +0100
@@ -541,9 +541,13 @@ void mutt_reflow_windows (void)
   {
     memcpy (MuttSidebarWindow, MuttIndexWindow, sizeof (mutt_window_t));
     MuttSidebarWindow->cols = SidebarWidth;
+    MuttIndexWindow->cols -= SidebarWidth;
 
-    MuttIndexWindow->cols -= SidebarWidth;
-    MuttIndexWindow->col_offset += SidebarWidth;
+    if (option (OPTSIDEBARONRIGHT)) {
+      MuttSidebarWindow->col_offset = COLS - SidebarWidth;
+    } else {
+      MuttIndexWindow->col_offset += SidebarWidth;
+    }
   }
 #endif
 }
diff -r 885f1d7126a4 -r 8867916fa2e2 init.h
--- a/init.h	Tue Jan 26 18:31:05 2016 +0000
+++ b/init.h	Thu Apr 21 02:32:18 2016 +0100
@@ -2784,6 +2784,11 @@ struct option_t MuttVars[] = {
   ** \fC<sidebar-prev-new>\fP command is similarly affected, wrapping around to
   ** the end of the list.
   */
+  { "sidebar_on_right", DT_BOOL, R_BOTH|R_REFLOW, UL OPTSIDEBARONRIGHT, 0 },
+  /*
+  ** .pp
+  ** When set, the sidebar will appear on the right-hand side of the screen.
+  */
   { "sidebar_short_path", DT_BOOL, R_SIDEBAR, OPTSIDEBARSHORTPATH, 0 },
   /*
   ** .pp
diff -r 885f1d7126a4 -r 8867916fa2e2 mutt.h
--- a/mutt.h	Tue Jan 26 18:31:05 2016 +0000
+++ b/mutt.h	Thu Apr 21 02:32:18 2016 +0100
@@ -455,6 +455,7 @@ enum
   OPTSIDEBARNEWMAILONLY,
   OPTSIDEBARNEXTNEWWRAP,
   OPTSIDEBARSHORTPATH,
+  OPTSIDEBARONRIGHT,
 #endif
   OPTSIGDASHES,
   OPTSIGONTOP,
diff -r 885f1d7126a4 -r 8867916fa2e2 sidebar.c
--- a/sidebar.c	Tue Jan 26 18:31:05 2016 +0000
+++ b/sidebar.c	Thu Apr 21 02:32:18 2016 +0100
@@ -487,10 +487,17 @@ static int draw_divider (int num_rows, i
 
   SETCOLOR(MT_COLOR_DIVIDER);
 
+  int col;
+  if (option (OPTSIDEBARONRIGHT)) {
+    col = 0;
+  } else {
+    col = SidebarWidth - delim_len;
+  }
+
   int i;
   for (i = 0; i < num_rows; i++)
   {
-    mutt_window_move (MuttSidebarWindow, i, SidebarWidth - delim_len);	//RAR 0 for rhs
+    mutt_window_move (MuttSidebarWindow, i, col);
     addstr (NONULL(SidebarDividerChar));
   }
 
@@ -501,21 +508,25 @@ static int draw_divider (int num_rows, i
  * fill_empty_space - Wipe the remaining Sidebar space
  * @first_row:  Window line to start (0-based)
  * @num_rows:   Number of rows to fill
- * @width:      Width of the Sidebar (minus the divider)
+ * @div_width:  Width in screen characters taken by the divider
+ * @num_cols:   Number of columns to fill
  *
  * Write spaces over the area the sidebar isn't using.
  */
-static void fill_empty_space (int first_row, int num_rows, int width)
+static void fill_empty_space (int first_row, int num_rows, int div_width, int num_cols)
 {
   /* Fill the remaining rows with blank space */
   SETCOLOR(MT_COLOR_NORMAL);
 
+  if (!option (OPTSIDEBARONRIGHT))
+    div_width = 0;
+
   int r;
   for (r = 0; r < num_rows; r++)
   {
-    mutt_window_move (MuttSidebarWindow, first_row + r, 0);	//RAR rhs
+    mutt_window_move (MuttSidebarWindow, first_row + r, div_width);
     int i;
-    for (i = 0; i < width; i++)
+    for (i = 0; i < num_cols; i++)
       addch (' ');
   }
 }
@@ -577,7 +588,12 @@ static void draw_sidebar (int num_rows, 
     else
       SETCOLOR(MT_COLOR_NORMAL);
 
-    mutt_window_move (MuttSidebarWindow, row, 0);
+    int col = 0;
+    if (option (OPTSIDEBARONRIGHT)) {
+      col = div_width;
+    }
+
+    mutt_window_move (MuttSidebarWindow, row, col);
     if (Context && Context->realpath &&
         !mutt_strcmp (b->realpath, Context->realpath))
     {
@@ -652,7 +668,7 @@ static void draw_sidebar (int num_rows, 
     row++;
   }
 
-  fill_empty_space (row, num_rows - row, w);
+  fill_empty_space (row, num_rows - row, div_width, w);
 }
 
 
@@ -676,7 +692,7 @@ void mutt_sb_draw (void)
 
   if (!Incoming)
   {
-    fill_empty_space (0, num_rows, SidebarWidth - div_width);
+    fill_empty_space (0, num_rows, div_width, num_cols - div_width);
     return;
   }
 
